DIPs/DIP1xxx-JS.md

```diff
@@ -0,0 +1,110 @@
+# The Deprecation Process
+
+| Field | Value |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | xxxx |
+| Review Count:   | 0 |
+| Author:         | Jack Stouffer |
+| Implementation: | N/A |
+| Status:         | Draft |
+
+## Abstract
+
+In order to incrementally improve D or it's standard library, it's often necessary to
+mark features or functions for future removal. This document proposes a standardized
+process for language maintainers to remove public features. This process would be 
+used across DMD, Druntime, and Phobos.
+
+## Rationale
+
+There is general disagreement on the best and/or accepted way to remove public
+features. Each deprecation ends up being handled slightly differently depending on
+who's handling the pull requests. Standardizing the process makes sure that
+deprecations are done very publicly and carefully, so as to minimize breakage and
+to provide clear fixes for user code.
+
+## Description
+
+The following lays out the rules to follow to deprecate a part of D.
+
+A symbol or feature must not be marked for removal on a specific date, but rather on a
+specific release. This allows users to easily know if upgrading will break their
+code or not.
+
+Users must be given at least 10 non-patch releases before the deprecated features
```

Comment:
This is part of why we haven't been using releases for deprecations but have been using dates. Dates are consistent. Releases haven't been. The number 10 is based on the idea that we're getting a new major release approximately every other month but probably slip occasionally, so it's about 2 years. If we're going to change the release cycle, then this number isn't appropriate. And if we're not going to be at least reasonably consistent about when releases are, IMHO, it makes no sense whatsoever to use releases as the measurement for the deprecation cycle.

That being said, I hate the idea of major releases only coming out every 6 months. Not only would that suck from the standpoint of getting new stuff, but that effectively means that deprecations can only be introduced twice a year. And while ideally, we would only be deprecating stuff rarely, that could seriously delay us being able to get rid of something that we decide that we need to get rid of.

Reply:
Releasing new versions has been a huge benefit to D IMO, and has done great things for other software like Firefox. I believe we move away from it at our detriment.

Reply:
Well, I can say for certain that in our case it is extremely unlikely to upgrade _major_ compiler version more often than once in a year (as the process itself usually takes several months across different projects). With that in mind 2 years is a very useful term because it ensures that you can't possibly miss deprecation of some feature by skipping 1 year worth of versions.

Actual amount of major versions that cover that period does not matter that much in this context (less often major versions are beneficial for other reasons but it not within scope of this DIP).